### PR TITLE
chore: change block switcher selection and linking behavior

### PR DESCRIPTION
**Pull request by harrysolovay**
Leaving this in draft because I've left some example code in `/start/getting-started/data-model/q/integration/js` and `/start/getting-started/hosting/q/integration/js` (for reviewers to play with). Please review and I'll follow up by removing example usage and taking this PR out of draft.

### CODE REVIEW

File: client/src/docs-ui/page/page.tsx [hunk 1]
```diff
@@ -183,117 +183,119 @@ export class DocsPage {
...
-                                  <docs-choose-anchor page={this.pageData} />
-                                )}
-                              </amplify-toc-contents>
-                            </amplify-sidebar-layout-main>
-                          ) : (
-                            [
-                              <amplify-sidebar-layout-main
-                                slot="main"
-                                class={mainStyle}
+          <amplify-block-switcher-provider>
```
Comment (jordanranz) [inline client/src/docs-ui/page/page.tsx:186]: I dislike this pattern a lot. We should definitely not leak this information into the page component. Is there a reason we can't just fire a block-switcher tab switch event in which other block switchers are listening to?

Comment (jordanranz) [reply client/src/docs-ui/page/page.tsx:227]: Is this still needed in the update to pageContext?

File: client/src/docs-ui/page/page.tsx [hunk 2]
```diff
@@ -50,6 +56,33 @@ export class DocsPage {
   @State() blendUniversalNav?: boolean;
   @State() sidebarStickyTop = getNavHeight("rem");
   @State() selectedFilters: Record<string, string | undefined> = {};
+  @State() selectedTabHeadings: SelectedTabHeadings = [];
+
+  setSelectedNewSelectedTabHeading: SetNewSelectedTabHeadings = (
+    tabHeading,
+  ) => {
+    // create temp array with `tabHeading` (the new highest priority) as first el
+    const temp = new Array<string>();
```
Comment (jordanranz) [inline client/src/docs-ui/page/page.tsx:65]: `temp` isn't a great variable name, please change this.

File: client/src/amplify-ui/block-switcher/block-switcher.tsx [hunk 3]
```diff
@@ -7,54 +7,70 @@ import {
...
-  tabLocalStorageKey: string;
-
   componentWillLoad() {
-    const children = Array.from(this.el.children);
-    const blocks = children.filter((child) => {
-      return child.matches(BLOCK_TAG_NAME);
-    });
+    // gather tab headings from child `amplify-block` attrs
+    this.tabHeadings = Array.from(this.el.children).reduce(
+      (acc, curr): string[] => {
```
Comment (jordanranz) [inline client/src/amplify-ui/block-switcher/block-switcher.tsx:34]: What is `acc`?

File: client/src/docs-ui/page/page.tsx [hunk 4]
```diff
@@ -50,6 +56,33 @@ export class DocsPage {
   @State() blendUniversalNav?: boolean;
   @State() sidebarStickyTop = getNavHeight("rem");
   @State() selectedFilters: Record<string, string | undefined> = {};
+  @State() selectedTabHeadings: SelectedTabHeadings = [];
+
+  setSelectedNewSelectedTabHeading: SetNewSelectedTabHeadings = (
```
Comment (jordanranz) [inline client/src/docs-ui/page/page.tsx:61]: `setSelectedNewSelectedTabHeading ` or `setNewSelectedTabHeadings`?

File: client/src/docs-ui/page/page.tsx (hunk 1 above)
Comment (harrysolovay) [reply client/src/docs-ui/page/page.tsx:227]: Good catch! No it isn't

File: client/src/amplify-ui/block-switcher/block-switcher.tsx (hunk 3 above)
Comment (harrysolovay) [reply client/src/amplify-ui/block-switcher/block-switcher.tsx:34]: The accumulator

**Pull request merged**